pkg/sink/codec/canal/canal_json_decoder.go

```diff
@@ -48,6 +56,8 @@ type batchDecoder struct {
 
 	upstreamTiDB *sql.DB
 	bytesDecoder *encoding.Decoder
+
+	tableInfoCache map[tableKey]*model.TableInfo
```

Comment:
after drop table and create table, table with same schema name and table name may have different TableInfo?

Reply:
cached item is deleted if receive table level DDL, and recreate a new table info.